cmd/object-handlers.go

```diff
@@ -402,7 +402,7 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		var oldKey, newKey []byte
 		sseCopyC := IsSSECopyCustomerRequest(r.Header)
 		sseC := IsSSECustomerRequest(r.Header)
-		if sseCopyC {
+		if sseCopyC && cpSrcDstSame {
```

Comment:
I am not sure why you added cpSrcDstSame here, Copy should work when we want to copy object & change its key at the same time. (I tested with AWS java SDK before)

Reply:
well the problem is extracting key means it is removing from request header so perhaps i need to simply add it back here.